tests/full_node_test.cpp

```diff
@@ -1547,30 +1547,87 @@ TEST_F(FullNodeTest, transaction_validation) {
 }
 
 TEST_F(FullNodeTest, light_node) {
-  auto node_cfgs = make_node_cfgs<20, true>(2);
+  auto node_cfgs = make_node_cfgs<10, true>(2);
   node_cfgs[0].is_light_node = true;
   node_cfgs[0].light_node_history = 10;
+  node_cfgs[0].dag_expiry_limit = 5;
+  node_cfgs[0].max_levels_per_period = 3;
+  node_cfgs[1].dag_expiry_limit = 5;
+  node_cfgs[1].max_levels_per_period = 3;
   auto nodes = launch_nodes(node_cfgs);
   uint64_t nonce = 0;
-  while (nodes[0]->getPbftChain()->getPbftChainSize() < 40) {
+  while (nodes[1]->getPbftChain()->getPbftChainSizeExcludingEmptyPbftBlocks() < 20) {
     Transaction dummy_trx(nonce++, 0, 2, 100000, bytes(), nodes[0]->getSecretKey(), nodes[0]->getAddress());
     // broadcast dummy transaction
-    nodes[0]->getTransactionManager()->insertTransaction(dummy_trx);
+    nodes[1]->getTransactionManager()->insertTransaction(dummy_trx);
     thisThreadSleepForMilliSeconds(200);
   }
   EXPECT_HAPPENS({10s, 1s}, [&](auto &ctx) {
     // Verify full node and light node sync without any issues
-    WAIT_EXPECT_EQ(ctx, nodes[0]->getPbftChain()->getPbftChainSize(), nodes[1]->getPbftChain()->getPbftChainSize())
+    WAIT_EXPECT_EQ(ctx, nodes[0]->getPbftChain()->getPbftChainSizeExcludingEmptyPbftBlocks(),
+                   nodes[1]->getPbftChain()->getPbftChainSizeExcludingEmptyPbftBlocks())
   });
-  uint32_t min_period_in_db = 0;
+  uint32_t non_empty_counter = 0;
   for (uint64_t i = 0; i < nodes[0]->getPbftChain()->getPbftChainSize(); i++) {
-    if (nodes[0]->getDB()->getPeriodDataRaw(i).size() > 0) {
-      min_period_in_db = i;
-      break;
+    const auto pbft_block = nodes[0]->getDB()->getPbftBlock(i);
+    if (pbft_block) {
```

Comment:
if (pbft_block && pbft_block->getPivotDagBlockHash() != NULL_BLOCK_HASH) {

Reply:
I want to count null block hashes as well. I just dont want to count the ones that are deleted from the db